pkg/api/v1/clients/factory/resource_client_factory.go

```diff
@@ -158,6 +158,10 @@ type ResourceClientFactory interface {
 	NewResourceClient(params NewResourceClientParams) (clients.ResourceClient, error)
 }
 
+type ResourceClientFactoryGetter interface {
+	ForCluster(cluster string, restConfig *rest.Config) ResourceClientFactory
```

Comment:
what's the purpose of this interface?
i'm a bit concerned as it's mixing concerns, getting a generic resource client factory vs setting a *rest.Config (should be kube-only)